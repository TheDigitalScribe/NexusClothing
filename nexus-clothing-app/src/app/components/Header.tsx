'use client'

import Link from 'next/link';
import { SearchBar } from './SearchBar';
import { MobileMenuToggle } from './MobileMenuToggle';
import { ThemeToggle } from './ThemeToggle';
import { useTheme } from './ThemeContext';
import { ThemeContextType } from '../types';
import { MagnifyingGlassIcon, UserIcon, ShoppingCartIcon } from '@heroicons/react/16/solid';

const NavBar: React.FC = () => {
  const { theme } = useTheme() as ThemeContextType;

  return (
    <nav className="py-4 px-6 md:px-12 shadow-md">
      <div className="container mx-auto flex justify-between items-center">

        {/* Logo */}
        <div className="flex-shrink-0">
          <Link href="/">
            <h1 className="text-lg md:text-xl font-bold transition-colors duration-200">NEXUS</h1>
          </Link>
        </div>

        {/* Navigation Links - hidden on mobile */}
        <div className="hidden md:flex space-x-8 items-center justify-center flex-grow text-lg">
          <Link href="/women-products" className="transition-colors duration-200">Women</Link>
          <Link href="/men-products" className="transition-colors duration-200">Men</Link>
          <Link href="/accessories" className="transition-colors duration-200">Accessories</Link>
          <Link href="/new-arrivals" className="transition-colors duration-200">New Arrivals</Link>
          <Link href="/sales" className="transition-colors duration-200">Sales</Link>
        </div>

        {/* Searchbar,  */}
        <div className="hidden md:flex items-center">
          <div>
            <SearchBar />
          </div>
          <div className="ml-4 space-x-4">
            <button>
              <ShoppingCartIcon className="w-6 h-6 transition-colors duration-200" />
            </button>
            <button>
              <UserIcon className="w-6 h-6 transition-colors duration-200" />
            </button>
          </div>
        </div>

        {/* Hamburger Menu */}
        <div className="flex md:hidden space-x-4">
          <MagnifyingGlassIcon className="w-6 h-6" />
          <MobileMenuToggle />
        </div>
      </div>
    </nav >
  )
}

export default NavBar;