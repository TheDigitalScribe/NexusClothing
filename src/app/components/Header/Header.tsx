"use client";

import React, { useState } from "react";
import Link from "next/link";
import { SearchBar } from "../SearchBar/SearchBar";
import { Switch } from "@/components/ui/switch"
import { MobileMenu } from "@/components/ui/mobile-menu";
import { MobileSearchBarDialog } from "@/components/ui/mobile-searchbar-dialog";
import { FaUser } from "react-icons/fa";
import { FaShoppingCart } from "react-icons/fa";

export const Header: React.FC = () => {
  const [user, setUser] = useState(false);

  const authPageLink = user ? "/user" : "/auth/login";

  return (
    <nav className="py-3 sm:py-4 px-4 sm:px-6 lg:px-20 shadow-md">
      <div className="container mx-auto flex justify-between items-center">

        {/* Logo */}
        <div className="flex-shrink-0">
          <Link href="/">
            <h1 className="text-lg sm:text-xl lg:text-2xl font-bold transition-colors duration-200 hover:text-blue-700">NEXUS</h1>
          </Link>
        </div>

        {/* Navigation Links */}
        <div className="hidden sm:flex space-x-8 items-center justify-center flex-grow">
          <Link href="/women-products" className="text-sm lg:text-lg transition-colors duration-200 hover:text-blue-700">Women</Link>
          <Link href="/men-products" className="text-sm lg:text-lg transition-colors duration-200 hover:text-blue-700">Men</Link>
          <Link href="/accessories" className="text-sm lg:text-lg transition-colors duration-200 hover:text-blue-700">Accessories</Link>
          <Link href="/sales" className="text-sm lg:text-lg transition-colors duration-200 hover:text-blue-700">Sales</Link>
        </div>

        {/* Searchbar, Shopping Cart and Account */}
        <div className="hidden md:flex items-center space-x-2 md:space-x-4">
          <div className="hidden md:flex">
            <SearchBar />
          </div>
          <div className="flex items-center space-x-2 md:space-x-4">
            <button aria-label="Shopping Cart Button">
              <FaShoppingCart className="w-6 h-6 transition-colors duration-200 hover:fill-blue-700/90" />
            </button>
            <Link href={authPageLink}>
              <button aria-label="Account Button" className="mt-1">
                <FaUser className="w-6 h-6 transition-colors duration-200 hover:fill-blue-700/90" />
              </button>
            </Link>
          </div>
          <div>
            <Switch />
          </div>
        </div>

        {/* Hamburger Menu and Search Icon for Mobile */}
        <div className="flex md:hidden space-x-4 items-center justify-center">
          <MobileSearchBarDialog />
          <MobileMenu />
        </div>
      </div>
    </nav>
  )
}