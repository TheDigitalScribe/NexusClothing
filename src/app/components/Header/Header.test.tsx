import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { Header } from './Header';

// Mock the search bar
jest.mock('../SearchBar', () => ({
  SearchBar: () => <div data-testid="search-bar">Search Bar</div>,
}));

// Mock the mobile menu toggle
jest.mock('../MobileMenuToggle/MobileMenuToggle', () => ({
  MobileMenuToggle: () => <div data-testid="mobile-menu-toggle">Mobile Menu Toggle</div>,
}));

// Mock next/link
jest.mock('next/link', () => {
  return ({ children, href }: { children: React.ReactNode; href: string }) => (
    <a href={href}>{children}</a>
  );
});

// Mock heroicons
jest.mock('@heroicons/react/16/solid', () => ({
  MagnifyingGlassIcon: () => <svg data-testid="magnifying-glass-icon" />,
  UserIcon: () => <svg data-testid="user-icon" />,
  ShoppingCartIcon: () => <svg data-testid="shopping-cart-icon" />,
}));

// Render test
describe('Header', () => {
  describe('Rendering', () => {
    it('renders the logo', () => {
      render(<Header />);
      expect(screen.getByText('NEXUS')).toBeInTheDocument();
    });

    // Render navigation links test
    it('renders all navigation links', () => {
      render(<Header />);
      const links = ['Women', 'Men', 'Accessories', 'New Arrivals', 'Sales'];
      links.forEach(link => {
        expect(screen.getByText(link)).toBeInTheDocument();
      });
    });

    // Render search bar test
    it('renders the search bar in desktop view', () => {
      render(<Header />);
      expect(screen.getByTestId('search-bar')).toBeInTheDocument();
    });

    // Render buttons test
    it('renders shopping cart and account buttons', () => {
      render(<Header />);
      expect(screen.getByLabelText('Shopping Cart Button')).toBeInTheDocument();
      expect(screen.getByLabelText('Account Button')).toBeInTheDocument();
    });
  });

  // Navigation tests
  describe('Navigation', () => {
    it('has correct href attributes for all links', () => {
      render(<Header />);
      const linkMappings = {
        'Women': '/women-products',
        'Men': '/men-products',
        'Accessories': '/accessories',
        'New Arrivals': '/new-arrivals',
        'Sales': '/sales'
      };
      Object.entries(linkMappings).forEach(([text, href]) => {
        const link = screen.getByText(text);
        expect(link).toHaveAttribute('href', href);
      });
    });

    // Logo link test
    it('logo links to home page', () => {
      render(<Header />);
      const logoLink = screen.getByText('NEXUS').closest('a');
      expect(logoLink).toHaveAttribute('href', '/');
    });
  });

  // Responsive design tests
  describe('Responsive Design', () => {
    it('renders mobile menu toggle on mobile view', () => {
      render(<Header />);
      expect(screen.getByTestId('mobile-menu-toggle')).toBeInTheDocument();
    });

    // Search icon test
    it('renders search icon on mobile view', () => {
      render(<Header />);
      expect(screen.getByTestId('magnifying-glass-icon')).toBeInTheDocument();
    });
  });

  // Accessibility tests
  describe('Accessibility', () => {
    it('has accessible buttons with proper aria-labels', () => {
      render(<Header />);
      expect(screen.getByLabelText('Shopping Cart Button')).toBeInTheDocument();
      expect(screen.getByLabelText('Account Button')).toBeInTheDocument();
    });
  });

  // Interactive tests
  describe('Interactive Elements', () => {
    it('shopping cart button is clickable', async () => {
      const user = userEvent.setup();
      render(<Header />);
      const cartButton = screen.getByLabelText('Shopping Cart Button');
      await user.click(cartButton);
    });

    // Account button test
    it('account button is clickable', async () => {
      const user = userEvent.setup();
      render(<Header />);
      const accountButton = screen.getByLabelText('Account Button');
      await user.click(accountButton);
    });

    // Snapshot test
    it('matches snapshot', () => {
      const { container } = render(<Header />);

      expect(container).toMatchSnapshot();
    });
  });
});
