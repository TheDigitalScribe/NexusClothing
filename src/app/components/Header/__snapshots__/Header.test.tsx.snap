// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Interactive Elements matches snapshot 1`] = `
<div>
  <nav
    class="py-4 px-6 md:px-12 shadow-md"
  >
    <div
      class="container mx-auto flex justify-between items-center"
    >
      <div
        class="flex-shrink-0"
      >
        <a
          href="/"
        >
          <h1
            class="text-lg md:text-xl font-bold transition-colors duration-200"
          >
            NEXUS
          </h1>
        </a>
      </div>
      <div
        class="hidden md:flex space-x-8 items-center justify-center flex-grow text-lg"
      >
        <a
          href="/women-products"
        >
          Women
        </a>
        <a
          href="/men-products"
        >
          Men
        </a>
        <a
          href="/accessories"
        >
          Accessories
        </a>
        <a
          href="/new-arrivals"
        >
          New Arrivals
        </a>
        <a
          href="/sales"
        >
          Sales
        </a>
      </div>
      <div
        class="hidden md:flex items-center"
      >
        <div>
          <div
            data-testid="search-bar"
          >
            Search Bar
          </div>
        </div>
        <div
          class="ml-4 space-x-4"
        >
          <button
            aria-label="Shopping Cart Button"
          >
            <svg
              data-testid="shopping-cart-icon"
            />
          </button>
          <button
            aria-label="Account Button"
          >
            <svg
              data-testid="user-icon"
            />
          </button>
        </div>
      </div>
      <div
        class="flex md:hidden space-x-4"
      >
        <svg
          data-testid="magnifying-glass-icon"
        />
        <div
          data-testid="mobile-menu-toggle"
        >
          Mobile Menu Toggle
        </div>
      </div>
    </div>
  </nav>
</div>
`;
